### ISSUE TITLE
Change openmp default to enabled

### ISSUE THREAD

**Issue by nouiz** [Theano/Theano#5683]
See: 

https://github.com/Theano/Theano/pull/5673

Now the original reason to disable it by default is gone. Maybe we should enable it by default now.

We should retest on windows and mac to be sure it work well as they don't always have openmp available.

**Comment by jakirkham** [comment]
Yeah Mac is not great with OpenMP. This would rely on a very old copy of `gcc` (i.e. 4.2.1) to get OpenMP support. AFAICT that compiler is basically unsupported.

Windows does have OpenMP though. So that should work ok.

Would this be optional? IOW could it be disabled?

**Comment by nouiz** [comment]
We have code that detect if the compiler support openmp and disable it if it isn't supported. Also, the user can set a Theano flag to disable it. In fact, it is the theano flag "openmp=False". Currently, we default to False. The auto detect code isn't used anymore, so we aren't sure it still work well.
